### ISSUE TITLE
[Break Change] Rename APIs new_with_* to with_*

### ISSUE THREAD

**Issue by TennyZhuang** [HdrHistogram/HdrHistogram_rust#119]
In [Rust's official naming guide](https://rust-lang.github.io/api-guidelines/naming.html), the recommanded constructor names should be `name` or `with_more_details`, and almost all stdlib APIs follow the guide.

This may introduce a break change so that a new major version should be released, feel free to close the issue if you don't think it's critical. This is only a suggestion.

**Comment by jonhoo** [comment]
Makes sense to me! I wouldn't remove the old ones though, just mark them as deprecated. Keeping around the new_ variants until we otherwise *need* to make a breaking change seems more reasonable.
